File: R/receptor_occupancy.R
ligand <- seq(0.01, 10, length.out = 200)
Kd <- 1.5
K <- 2
n <- 3

occupancy <- ligand / (Kd + ligand)
hill_response <- ligand^n / (K^n + ligand^n)

par(mfrow = c(1, 2))

plot(ligand, occupancy, type = "l", lwd = 2,
     xlab = "Ligand concentration",
     ylab = "Fraction occupied",
     main = "Receptor occupancy")

plot(ligand, hill_response, type = "l", lwd = 2,
     xlab = "Ligand concentration",
     ylab = "Normalized response",
     main = "Hill-type signaling response")

File: python/receptor_occupancy.py
import numpy as np
import matplotlib.pyplot as plt

L = np.linspace(0.01, 10, 300)
Kd = 1.5
K = 2.0
n = 3

occupancy = L / (Kd + L)
hill_response = L**n / (K**n + L**n)

plt.plot(L, occupancy, label="Occupancy")
plt.plot(L, hill_response, label="Hill response")
plt.xlabel("Ligand concentration")
plt.ylabel("Normalized response")
plt.title("Receptor occupancy and cooperative response")
plt.legend()
plt.show()
